packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemPath.kt

```diff
@@ -62,23 +62,13 @@ abstract class FileSystemPath(var uri: Uri) : SharedObject() {
       throw UnableToDeleteException("uri '${file.uri}' does not exist")
     }
     if (file.isDirectory()) {
-      file.listFilesAsUnified().forEach { child ->
-        if (child.isDirectory()) {
-          // Recursively delete subdirectories
-          if (uri.isContentUri) {
-            SAFDocumentFile(appContext?.reactContext ?: throw Exception("No context"), child.uri).delete()
-          } else {
-            JavaFile(child.uri).delete()
-          }
-        } else {
-          if (!child.delete()) {
-            throw UnableToDeleteException("failed to delete '${child.uri}'")
-          }
-        }
+      if (!file.deleteRecursively()) {
+        throw UnableToDeleteException("failed to delete '${file.uri}'")
```

Comment:
This exception could be more descriptive

Reply:
only thing I know about this error is boolean status, so not sure what more we could say here